dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/domain/BasePersistenceStrategy.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2019 Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.domain
+
+import com.datadog.android.core.internal.data.DataMigrator
+import com.datadog.android.core.internal.data.Reader
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.data.file.DeferredWriter
+import com.datadog.android.core.internal.data.file.FileOrchestrator
+import com.datadog.android.core.internal.data.file.FileReader
+import com.datadog.android.core.internal.data.file.ImmediateFileWriter
+import java.io.File
+
+internal abstract class BasePersistenceStrategy<T : Any>(
```

Comment:
I'd rename that `FilePersistenceStrategy`, and also let the `LogFileStrategyTest` become `FilePersistenceStrategyTest` and test this class. After all the LogFileStrategy doesn't do anything anymore ;)